src/main/scala/com/fotolog/redis/codecs/RedisCommandEncoder.scala

```diff
@@ -0,0 +1,40 @@
+package com.fotolog.redis.codecs
+
+import com.fotolog.redis.connections.ResultFuture
+import io.netty.buffer.ByteBuf
+import io.netty.channel.ChannelHandler.Sharable
+import io.netty.channel.ChannelHandlerContext
+import io.netty.handler.codec.MessageToByteEncoder
+
+/**
+  * @author Yaroslav Derman <yaroslav.derman@gmail.com>.
+  *         created on 02.03.2017.
+  */
+@Sharable
+private[redis] class RedisCommandEncoder extends MessageToByteEncoder[ResultFuture] {
+
+  import com.fotolog.redis.connections.Cmd._
+
+  override def encode(ctx: ChannelHandlerContext, msg: ResultFuture, out: ByteBuf): Unit = {
+    binaryCmd(msg.cmd.asBin, out)
+  }
+
+  private def binaryCmd(cmdParts: Seq[Array[Byte]], out: ByteBuf) = {
+    val params = new Array[Array[Byte]](3 * cmdParts.length + 1)
```

Comment:
This is redundant: you shouldn't create arrays here, vut write directly to the `out`, this can increase performance a lot

Reply:
How do you think. What is it better 
```
out.writeBytes(("$" + p.length + "\r\n").getBytes) // len of the chunk
out.writeBytes(p)
out.writeBytes(EOL)
```
or
```
out.writeBytes(("$" + p.length + "\r\n").getBytes ++ p ++ EOL)
```
?

Reply:
done